### PR TITLE
NewOmega Application

### PR DESCRIPTION
**Pull request by celrisen**
# Grant Application Checklist

- [X] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [X] A BTC or Ethereum address for the payment of the milestones is provided inside the application.  
- [X] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [X] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants at the time of submission.
- [X] The initial PR contains only one commit (squash if needed before submitting your PR).
- [X] The grant will only be announced once we successfully [delivered](https://github.com/w3f/Grant-Milestone-Delivery) the first milestone.

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Sounds like an interesting project. Just for clarification you are referring to the following contracts her: https://github.com/WiktorStarczewski/newomega.trinity/blob/master/hardhat/contracts, correct? Maybe you can link to them in the deliveries. This makes it easier for us to evaluate the deliveries later. Also it seems like you are putting a lot of information on-chain, which might make the game crazy expensive. Have you thought about integrating an additional second layer  solution like for example https://www.celer.network/?

**Comment by celrisen** [comment]
> Thanks for the application. Sounds like an interesting project. Just for clarification you are referring to the following contracts her: https://github.com/WiktorStarczewski/newomega.trinity/blob/master/hardhat/contracts, correct? Maybe you can link to them in the deliveries. This makes it easier for us to evaluate the deliveries later. Also it seems like you are putting a lot of information on-chain, which might make the game crazy expensive. Have you thought about integrating an additional second layer solution like for example https://www.celer.network/?

Hi David, great question. So what's currently implemented, in the "attacking" smart contract is storing game state and fight replays on the chain (state directly in storage, replays in event logs). That is obviously costly, but mostly due to the storing and processing of the replays - if I just calculate the outcome and store basic win/lose information, I can get the gas costs of that contract down to something that starts to be acceptable from game economics point of view. (And I could do something like regenerate the replay itself in the frontend from say, the pseudo-random seed that was used, right now a naive block stamp, but obviously could be Oracle, to eliminate the need for logs storage). Of course the definition of acceptable here perhaps does not extend to ETH Mainnet right now, thats a known problem, but there are (or almost are/about to be) networks out there that would make this calculation cheap enough for the game economics to make sense (and of course here's hoping Polkadot can improve on the current ETH design quite significantly). Not to mention additional optimisation rounds that the contracts would receive over time.

So to sum up, I feel like I'd like to explore more the L1 angle, but certainly keeping an open mind.

I've also updated the milestones with the GitHub links, as you requested.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the quick response. Makes sense and I’m generally interested in supporting your application, but I think it’s currently a little bit expensive. Could you potentially remove milestone 3 for now, since we can always sign a follow-up grant once you delivered the first PoC. Furthermore, would it be possible to reduce the price for milestone 1 and 2 a little bit?  This way it would be easier to accept for us. And I guess part of the milestone 2 deliveries is the UI + the integration of the polkadot.js extension. “Integrate game client with polkadot” might currently be a little bit misleading, since you won’t integrate into polkadot (for example as a parachain).

**Comment by celrisen** [comment]
> Thanks for the quick response. Makes sense and I’m generally interested in supporting your application, but I think it’s currently a little bit expensive. Could you potentially remove milestone 3 for now, since we can always sign a follow-up grant once you delivered the first PoC. Furthermore, would it be possible to reduce the price for milestone 1 and 2 a little bit? This way it would be easier to accept for us. And I guess part of the milestone 2 deliveries is the UI + the integration of the polkadot.js extension. “Integrate game client with polkadot” might currently be a little bit misleading, since you won’t integrate into polkadot (for example as a parachain).

Ok, makes sense, I'd certainly be interested in further development after the two milestones, if we decide we're a good fit. To that end, I've removed milestone 3 for now and tweaked the price of 1 and 2. Let me know what you think.

**Review by Noc2 — APPROVED** [review]
Thanks again for the update. I will share it with the rest of the team.

**Review by BenWhiteJam — APPROVED** [review]

**Comment by Lederstrumpf** [comment]
Hi @celrisen, cool looking demo so far! After I authorize Ropsten payments, the interface remains on "Waiting for blockchain". Which aspects of your demo are supposed to work? Loading assets takes ages, but I can watch the demos under "training" and "logs", which is good enough for me at this stage of the PoC. Anyway - no need to fix this since it's an Ethereum demo ;)

Approved from me too; looking forward!

**Review by Lederstrumpf — APPROVED** [review]
